lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1037,6 +1037,84 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
     return DITy;
   }
 
+  /// Creates debug info for a generic struct with archetypes (eg.:
+  /// Pair<τ_0_0, τ_0_1>). For types with unsubstituted generic type paramaters,
+  /// debug info generation doesn't attempt to emit the size and aligment of
+  /// the type, as those are all dependent on substituting the type parameters
+  /// in. It also doesn't attempt to emit the offset of the members for the same
+  /// reason.
+  llvm::DICompositeType *createUnsubstitutedGenericStructType(
+      DebugTypeInfo DbgTy, NominalTypeDecl *Decl, Type UnsubstitutedType, Type SubstitutedType,
+      llvm::DIScope *Scope, llvm::DIFile *File, unsigned Line,
+      llvm::DINode::DIFlags Flags, llvm::DIType *DerivedFrom,
+      unsigned RuntimeLang, StringRef UniqueID) {
+    // FIXME: ideally, we'd like to emit this type with no size and alignment at
+    // all (instead of emitting them as 0). Fix this by changing DIBuilder to
+    // allow for struct types that have optional size and alignment.
+    
+    StringRef Name = Decl->getName().str();
+
+    // Forward declare this first because types may be recursive.
+    auto FwdDecl = llvm::TempDIType(DBuilder.createReplaceableCompositeType(
+        llvm::dwarf::DW_TAG_structure_type, Name, Scope, File, Line,
+        llvm::dwarf::DW_LANG_Swift, 0, 0, Flags, UniqueID));
+
+#ifndef NDEBUG
+    if (UniqueID.empty())
+      assert(!Name.empty() &&
+             "no mangled name and no human readable name given");
+    else
+      assert((UniqueID.startswith("_T") ||
+              UniqueID.startswith(MANGLING_PREFIX_STR)) &&
+             "UID is not a mangled name");
+#endif
+
+    auto TH = llvm::TrackingMDNodeRef(FwdDecl.get());
+    DITypeCache[DbgTy.getType()] = TH;
+    // Collect the members.
+    SmallVector<llvm::Metadata *, 16> Elements;
+    for (VarDecl *VD : Decl->getStoredProperties()) {
+      auto memberTy =
+          UnsubstitutedType->getTypeOfMember(IGM.getSwiftModule(), VD);
+      llvm::DIType *DITy = nullptr;
+      if (llvm::isa<swift::GenericTypeParamType>(memberTy)) {
```

Comment:
Just `isa<GenericTypeParamType>` should work. However, I don't see what is special about generic parameter types here. You can also have a member type (T.A), wouldn't that be the same as a generic parameter?

Reply:
You're right, this was incorrect.